### PR TITLE
Ignore handling logic for importlib_metadata (mypy, tests/plugins)

### PR DESCRIPTION
**Pull request by khalidmammadov**
Part of https://github.com/apache/airflow/issues/19891

---
**^ Add meaningful description above**

Read the **[Pull Request Guidelines](https://github.com/apache/airflow/blob/main/CONTRIBUTING.rst#pull-request-guidelines)** for more information.
In case of fundamental code change, Airflow Improvement Proposal ([AIP](https://cwiki.apache.org/confluence/display/AIRFLOW/Airflow+Improvements+Proposals)) is needed.
In case of a new dependency, check compliance with the [ASF 3rd Party License Policy](https://www.apache.org/legal/resolved.html#category-x).
In case of backwards incompatible changes please leave a note in [UPDATING.md](https://github.com/apache/airflow/blob/main/UPDATING.md).

### CODE REVIEW

File: tests/plugins/test_plugins_manager.py [hunk 1]
```diff
@@ -30,24 +30,14 @@
 from tests.test_utils.config import conf_vars
 from tests.test_utils.mock_plugins import mock_plugin_manager
 
-importlib_metadata_string = 'importlib_metadata'
-
-try:
+if sys.version_info[:2] >= (3, 8):
```
Comment (potiuk) [inline tests/plugins/test_plugins_manager.py]: I think we have PY38 constant for that in `airflow':  @khalidmammadov . 

Also @ashb - I recall discussions we had about it, I think it makes sense to based it on  Python version

**Comment by uranusjr** [comment]
The problem is that `importlib.metadata` is unstable (the official term is _provisional_) before 3.10, and not everything in `importlib_metadata` is available in `importlib.metadata`. So if we want to switch to, say, the 3.9 version of `importlib.metadata`, we’d bump the `python_version` marker in `setup.cfg`, but after this change we’d also need to change a ton of if-imports due to that change. Try-imports are easier because it’d always try the third-party version first, regardless of the Python version.

If we want to change try-imports to if-imports, it’s probably better to always use `importlib_metadata` for all versions prior to 3.10, when the stdlib `importlib.metadata` enters stable state.

Also I believe `PY38` would not work because Mypy uses AST detection to perform type-time conditional magic, and would not understand what `PY38` means correctly.

File: tests/plugins/test_plugins_manager.py (hunk 1 above)
Comment (ashb) [reply tests/plugins/test_plugins_manager.py]: Yeah, someone (i forget who, maybe Martjin Peiters? sp?) recommended the approach TP says -- always try the third part first, as that API is stable and predictable.)

**Comment by potiuk** [comment]
> The problem is that `importlib.metadata` is unstable (the official term is _provisional_) before 3.10

That would explain some of the behaviours seen before :)

**Comment by khalidmammadov** [comment]
> The problem is that `importlib.metadata` is unstable (the official term is _provisional_) before 3.10, and not everything in `importlib_metadata` is available in `importlib.metadata`. So if we want to switch to, say, the 3.9 version of `importlib.metadata`, we’d bump the `python_version` marker in `setup.cfg`, but after this change we’d also need to change a ton of if-imports due to that change. Try-imports are easier because it’d always try the third-party version first, regardless of the Python version.
> 
> If we want to change try-imports to if-imports, it’s probably better to always use `importlib_metadata` for all versions prior to 3.10, when the stdlib `importlib.metadata` enters stable state.
> 
> Also I believe `PY38` would not work because Mypy uses AST detection to perform type-time conditional magic, and would not understand what `PY38` means correctly.

I will revert and try to fix MyPy issues different way

**Review by potiuk — APPROVED** [review]

**Comment by ashb** [comment]
@khalidmammadov Please read https://cbea.ms/git-commit/ (linked to from our contributing guide) and try to give more useful commit messages in future.

**Comment by khalidmammadov** [comment]
> @khalidmammadov Please read https://cbea.ms/git-commit/ (linked to from our contributing guide) and try to give more useful commit messages in future.

Do you mean commit message or PR description/title?

**Pull request merged**